<launch>
  <arg name="load_manager" />
  <arg name="manager_name" />
  <arg name="manager_threads" />
  <arg name="camera_name" />

  <node if="$(arg load_manager)" pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen">
    <param name="num_worker_threads" value="$(arg manager_threads)"/>
  </node>

  <!-- tf Remapping -->
  <node pkg="camera_tracking" type="tf_rebroadcaster" name="tf_rebroadcaster" output="screen">
    <param name="prefix" value="$(arg camera_name)"/>
  </node>
  <node pkg="camera_tracking" type="camera_origin" name="camera_origin" output="screen">
    <param name="camera_name" value="$(arg camera_name)"/>
    <param name="origin_id" value="tag_0"/>
  </node>

  <!-- De-Bayer / rectify -->
  <include file="$(find image_proc)/launch/image_proc.launch">
    <arg name="manager" value="$(arg manager_name)"/>
  </include>

  <!-- Apriltag detection -->
  <node pkg="nodelet" type="nodelet" name="apriltag" args="load apriltag_ros/ContinuousDetector $(arg manager_name)" output="screen">
    <remap from="/tf" to="tf"/>
    <param name="camera_frame" type="str" value="$(arg camera_name)"/>
    <param name="publish_tag_detections_image" type="bool" value="true"/>
    <rosparam command="load" file="$(find camera_tracking)/config/apriltag_settings.yaml"/>
    <rosparam command="load" file="$(find camera_tracking)/config/tags.yaml"/>
  </node>
</launch>
